enhancements/storage/csi-migration.md

```diff
@@ -0,0 +1,145 @@
+
+With that in mind, we propose to:
+
+1. Create two [new FeatureSets](https://github.com/openshift/api/blob/master/config/v1/types_feature.go#L25-L43) to support CSI migration: `CSIMigrationNode` and `CSIMigrationControlPlane`.
+   * Both *FeatureSets* contain the same feature gates: `CSIMigration`, `CSIMIgrationAWS`, `CSIMigrationGCE`, `CSIMigrationAzureDisk`, `CSIMigrationAzureFile`, `CSIMigrationvSphere`, `CSIMigrationOpenStack`.
+   * However, only control-plane operators will reacto to the `CSIMigrationControlPlane` *FeatureSet*. The machine-config-operator (MCO) will **ignore** it.
+   * On the other hand, only MCO will react to the `CSIMigrationNode` *FeatureSet*. Control-plane operators will ignore it.
+2. To enable CSI Migration for any in-tree plugin, the cluster administrator should:
+   * Add the `CSIMigrationControlPlane` *FeatureSet* to the `featuregates/cluster` object:
+   ```shell
+   $ oc edit featuregates/cluster
+   (...)
+   spec:
+     featureSet: CSIMigrationControlPlane
+   ```
+   * Once all control-plane components have restart, add the `CSIMigrationNode` *FeatureSet* to the `featuresgates/cluster` object:
+   ```shell
+   $ oc edit featuregates/cluster
+   (...)
+   spec:
+     featureSet: CSIMigrationControlPlane, CSIMigrationNode
+   ```
+3. To disable CSI migration, the cluster administrator should perform the same steps in the opposite order:
+   * Add the `CSIMigrationNode` *FeatureSet* to the `featuregates/cluster` object.
+   * Wait for all Nodes to be drained and restarted.
+   * Add the `CSIMigrationControlPlane` *FeatureSet* to the `featuregates/cluster` object.
+4. It is the **responsability of the cluster administrator** to guarante the ordering described above is respected.
+
+### GA
+
+Once CSI migration reaches GA in upstream, the associated feature gates will be enabled by default and the features will not be optional anymore. As a result, CSI migration will be enabled by default in OCP as wel, and there will not be an option to disable it.
+
+In addition that, the *FeatureSets* created to handle the Tech Preview feature will no longer be operatinal. In that case, those *FeatureSets* should be ignored by operators.
+
+As for the required ordering described above, the [upgrade order](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/upgrades.md#generalized-ordering) performed by CVO during a cluster upgrade will take care of applying the feature gates in the correct order.
+
+#### Limitations
+
+The limitations of this approach lies on the downgrade process.
+
+In OCP, a downgrade is fundamentally a regular upgrade to an older version. That means that CVO downgrades components in the same order as it upgrades them: control-plane first, nodes later.
```

Comment:
I would note that this violates upstream version skew policy. A new kubelet must never run with an older API server or controller manager. Complicated downgrade with CSI migration is a direct consequence of this violation. If OCP did downgrade right, no extra actions would be necessary.

Reply:
Added